R/helpers_data.R

```diff
@@ -0,0 +1,235 @@
+#' Helper Function to Blind Plot Data
+#'
+#' @param df (`GeneralData`)\cr The data to be blinded
+#' @param blind (`flag`)\cr Should the data be blinded?
+#' @param has_placebo (`flag`)\cr Does the data contain a placebo dose?
+#' @param pbo_dose (`positive_number`)\cr The dose to be taken as placebo.
+#' Ignored if `has_placebo` is `FALSE`
+#' @returns The blinded data
+h_blind_plot_data <- function(df, blind, has_placebo, pbo_dose) {
+  if (blind) {
+    # This is to blind the data.
+    # For each cohort, all DLTs are assigned to the first subjects in the cohort.
+    # In addition, the placebo (if any) is set to the active dose level for that
+    # cohort.
+    # Notice: dapply reorders records of df according to the lexicographic order
+    # of cohort.
+    df <- dapply(df, f = ~cohort, FUN = function(coh) {
+      coh$toxicity <- sort(coh$toxicity, decreasing = TRUE)
+      coh$dose <- max(coh$dose)
+      coh
+    })
+  } else if (has_placebo) {
+    # Placebo will be plotted at y = 0 level.
+    df$dose[df$dose == pbo_dose] <- 0
+  }
+  df
+}
+
+# h_plot_data_df ----
+
+## generic ----
+
+#' Helper Function for the Plot Method of subclasses of [`GeneralData`]
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' A method that transforms [`GeneralData`]  objects into a `tibble` suitable for
+#' plotting with `ggplot2` methods
+#'
+#' @param data (`GeneralData`)\cr object from which data is extracted and converted
+#' into a data frame.
+#' @param ... further arguments passed to class-specific methods.
+#' @return `data.frame` containing columns for patient, cohort, dose and toxicity grade
+#' @aliases h_plot_data_df
+#'
+setGeneric(
+  name = "h_plot_data_df",
+  def = function(data, ...) standardGeneric("h_plot_data_df"),
+  valueClass = "data.frame"
+)
+
+# Data ----
+
+#' Helper Function for the Plot Method of [`Data`]
+#'
+#' @param data (`Data`)\cr object from which data is extracted and converted
+#'   into a data frame.
+#' @param blind (`flag`)\cr should data be blinded?
+#'   If `TRUE`, then for each cohort, all DLTs are assigned to the first
+#'   subjects in the cohort. In addition, the placebo (if any) is set to the
+#'   active dose level for that cohort.
+#' @param legend (`flag`)\cr Display the legend for the toxicity categories
+#' @param ... further arguments passed to `data.frame` constructor.
+#'   It can be e.g. an extra `column_name = value` pair based on a slot
+#'   from `x` (which in this case might be a subclass of `Data`)
+#'   which does not appear in `Data`.
+#' @return A [`data.frame`] object with columns patient, ID, cohort, dose and
```

Comment:
```suggestion
#' @return A `data.frame` object with columns patient, ID, cohort, dose and
```
(no need to link to the documentation for this well known class)